libcfnet/protocol.c

```diff
@@ -88,3 +89,88 @@ Seq *ProtocolOpenDir(AgentConnection *conn, const char *path)
 
     return seq;
 }
+
+bool ProtocolGet(AgentConnection *conn, const char *remote_path,
+                 const char *local_path, const uint32_t file_size, int perms)
+{
+    assert(conn != NULL);
+    assert(remote_path != NULL);
+    assert(local_path != NULL);
+    assert(file_size != 0);
+
+    perms = (perms == 0) ? CF_PERMS_DEFAULT : perms;
+
+    unlink(local_path);
+    FILE *file_ptr = safe_fopen_create_perms(local_path, "w", perms);
+    if (file_ptr == NULL)
+    {
+        return false;
```

Comment:
Does safe_fopen_create_perms() log what went wrong? Seems like it does in most cases in safe_open_create_perms() so probably OK to just return false with no logging here.

Reply:
No, I need to add better log messages here. : )